v2/cmd/integration-test/custom-dir.go

```diff
@@ -0,0 +1,34 @@
+package main
+
+import (
+	"os"
+
+	"github.com/projectdiscovery/nuclei/v2/pkg/testutils"
+)
+
+func getCustomConfigsDir() string {
+	temp := os.TempDir()
+	return temp
+}
+
+type customConfigDirTest struct{}
+
+var customConfigDirTestCases = map[string]testutils.TestCase{
+	"dns/cname-fingerprint.yaml": &customConfigDirTest{},
+}
+
+// Execute executes a test case and returns an error if occurred
+func (h *customConfigDirTest) Execute(filePath string) error {
+	defer os.RemoveAll(getCustomConfigsDir())
+
+	var routerErr error
+
+	results, err := testutils.RunNucleiTemplateAndGetResults(filePath, "8x8exch02.8x8.com", debug, "-config-directory", getCustomConfigsDir())
+	if err != nil {
+		return err
+	}
+	if routerErr != nil {
+		return routerErr
+	}
+	return expectResultsCount(results, 1)
```

Comment:
We need also to ensure that in `customTempDirectory` the following files are created:
```
.nuclei-ignore
.templates-config.json
config.yaml
```
So the test should also implement something like:
```go
files, err := os.ReadDir(customTempDirectory)
if err != nil {
	return err
}
// 3 files are expected:
// - .nuclei-ignore
// - .templates-config.json
// - config.yaml
var fileNames []string
for _, file := range files {
	fileNames = append(fileNames, file.Name())
}
return expectResultsCount(fileNames, 3)
```

Reply:
Done